deep_quoridor/src/agents/alphazero/alphazero.py

```diff
@@ -546,56 +563,88 @@ def _log_action(
+                visit_probs = visit_probs ** (1.0 / temperature)
+                visit_probs = visit_probs / np.sum(visit_probs)
 
-        temperature = self.initial_temperature
-        if self.params.drop_t_on_step is not None and game.completed_steps >= self.params.drop_t_on_step:
-            temperature = 0
+            # Sample from probability distribution
+            best_child = np.random.choice(root_children, p=visit_probs)
+            action = best_child.action_taken
+            actions.append((game_idx, self.action_encoder.action_to_index(action)))
 
-        if temperature == 0.0:
-            max_value = np.max(visit_probs)
-            visit_probs = np.array([1.0 if v == max_value else 0.0 for v in visit_probs])
-            visit_probs /= np.sum(visit_probs)
-        else:
-            visit_probs = visit_probs ** (1.0 / temperature)
-            visit_probs = visit_probs / np.sum(visit_probs)
-
-        # Sample from probability distribution
-        best_child = np.random.choice(root_children, p=visit_probs)
-        action = best_child.action_taken
-
-        if self.params.penalized_visited_states:
-            self.visited_states.add(QuoridorKey(best_child.game))
-        # Store training data if in training mode
-        if self.params.training_mode:
-            # Convert visit counts to policy target (normalized)
-            policy_target = self.action_encoder.get_action_mask_template()
-            for child in root_children:
-                action_index = self.action_encoder.action_to_index(child.action_taken)
-                policy_target[action_index] = child.visit_count / visit_counts_sum
-            self.store_training_data(game, policy_target, player)
-
-        return self.action_encoder.action_to_index(action)
-
-    def store_training_data(self, game, mcts_policy, player):
+            # TODO: this is disabled with multiple ids because we would need to keep multiple visited states sets,
+            # one per game. Is it worth implementing?
+            if len(observations_with_ids) == 1 and self.params.penalized_visited_states:
+                self.visited_states.add(QuoridorKey(best_child.game))
+
+            # Store training data if in training mode
+            if self.params.training_mode:
+                # Convert visit counts to policy target (normalized)
+                policy_target = self.action_encoder.get_action_mask_template()
+                for child in root_children:
+                    action_index = self.action_encoder.action_to_index(child.action_taken)
+                    policy_target[action_index] = child.visit_count / visit_counts_sum
+                self.store_training_data(game, policy_target, player, game_idx)
+
+        return actions
+
+    def get_action(self, observation) -> int:
+        action_batch = self.get_action_batch([(0, observation)])
+        return action_batch[0][1]
+
+    def store_training_data(self, game, mcts_policy, player, game_idx):
```

Comment:
Added game_idx, because the training data for each parallel game needs to be store separately, so that when we finish we set the right value.  (Actually now that I think about it I could have stored everything in 1 replay buffer and add the game_idx as an entry, but not sure if it makes anything better anyway)